src/app/features/collection/services/collection-api.service.spec.ts

```diff
@@ -0,0 +1,196 @@
+
+  const responseMock = { data: {}, message: 'message' };
+
+  beforeEach(() => {
+    httpClientMock = mock<HttpClient>();
+    httpClientMock.post.mockReturnValue(of(responseMock));
+    httpClientMock.get.mockReturnValue(of(responseMock));
+    httpClientMock.delete.mockReturnValue(of(responseMock));
+    httpClientMock.patch.mockReturnValue(of(responseMock));
+
+    validationServiceMock = mock<ValidationService>();
+    validationServiceMock.validate.mockReturnValue(responseMock);
+
+    service = classWithProviders({
+      token: CollectionApiService,
+      providers: [
+        {
+          provide: HttpClient,
+          useValue: httpClientMock,
+        },
+        {
+          provide: ValidationService,
+          useValue: validationServiceMock,
+        },
+      ],
+    });
+  });
+
+  describe('getCollectionInfo$', () => {
+    it('should trigger "get" method with the correct args', () => {
+      const contextMock = new HttpContext().set(AuthConstants.skipLoadingContextToken, true);
+
+      service.getCollectionInfo$().pipe(take(1)).subscribe();
+
+      expect(httpClientMock.get).toHaveBeenCalledWith(environment.endpoints.collection.getCollectionInfo, {
+        context: contextMock,
+      });
+    });
+
+    it('should validate the response', () => {
+      service.getCollectionInfo$().pipe(take(1)).subscribe();
+
+      expect(validationServiceMock.validate).toHaveBeenCalledWith(
+        CollectionSchemas.collectionInfoResponseDto,
+        responseMock,
+      );
+    });
+
+    it('should return the data from the response', done => {
+      service
+        .getCollectionInfo$()
+        .pipe(take(1))
+        .subscribe(res => {
+          expect(res).toBe(responseMock.data);
+
+          done();
+        });
+    });
+  });
+
+  describe('getCardsByRarity$', () => {
+    it('should trigger "get" method with the correct args', () => {
+      const contextMock = new HttpContext().set(AuthConstants.skipLoadingContextToken, true);
+      const rarity = 'SSR';
+
+      service.getCardsByRarity$(rarity).pipe(take(1)).subscribe();
+
+      expect(httpClientMock.get).toHaveBeenCalledWith(
+        `${environment.endpoints.collection.getByRarity}?rarity=${rarity}`,
+        { context: contextMock },
+      );
+    });
+
+    it('should validate the response', () => {
+      service.getCardsByRarity$('R').pipe(take(1)).subscribe();
+
+      expect(validationServiceMock.validate).toHaveBeenCalledWith(
+        CollectionSchemas.collectionCardsResponseDto,
+        responseMock,
+      );
+    });
+
+    it('should return the cards from the response', done => {
+      const cardsMock = [{}];
+      const resnonseWithCardsMock = { data: { cards: cardsMock } };
+
+      validationServiceMock.validate.mockReturnValueOnce(resnonseWithCardsMock);
+
+      service
+        .getCardsByRarity$('R')
+        .pipe(take(1))
+        .subscribe(res => {
+          expect(res).toBe(cardsMock);
+
+          done();

```

Comment:
this test should work without done. In case if it does not work for you - check mocking part

Reply:
If you don't use done, the test will always pass regardless of the result of expect().toBe()